### ISSUE TITLE
Linking fails

### ISSUE THREAD

**Issue by Armagetron** [gammu/gammu#159]
I have an application that should connect to the smsd and inject an sms (pretty much the same as gammu-smsd-inect) but directly from my C++ code.

I tried the following: just linking to libGammu. Then the SMSD_* function symbols are not resolved.

When I try to link just to libgsmsd this also fails because the Gammu symbols are not resolved:
```
/usr/bin/ld: /tmp/ccss1QAF.o: undefined reference to symbol 'GSM_GetGlobalDebug'
//usr/local/lib/libGammu.so.7: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```
Linking to libgsmsd and libGammu (in this order) gives this error:
```
/tmp/ccgpHCRR.o: In function `SendSMS(std::string)':
bosmon.cpp:(.text+0x650): undefined reference to `SMSD_NewConfig(char const*)'
bosmon.cpp:(.text+0x668): undefined reference to `SMSD_ReadConfig(char const*, _GSM_SMSDConfig*, int)'
bosmon.cpp:(.text+0x690): undefined reference to `SMSD_FreeConfig(_GSM_SMSDConfig*)'
bosmon.cpp:(.text+0x6c0): undefined reference to `SMSD_InjectSMS(_GSM_SMSDConfig*, GSM_MultiSMSMessage*, char*)'
bosmon.cpp:(.text+0x6e8): undefined reference to `SMSD_FreeConfig(_GSM_SMSDConfig*)'
bosmon.cpp:(.text+0x734): undefined reference to `SMSD_FreeConfig(_GSM_SMSDConfig*)'
collect2: error: ld returned 1 exit status
```
Can you give me a hint?

**Comment by nijel** [comment]
You should definitely link with both of them. However if you include directly gammu-smsd.h it lacks C namespace declaration, what makes it fails to link with CPP.

**Comment by Armagetron** [comment]
The revision bfce1a3635b8d0901bdba03f9906cc6e41062340 does not compile, because the cpp-sample directory is missing.

However this does not fix it for me. Still got the error when linking libgsmsd and libGammu.

**Comment by nijel** [comment]
I've fixed it in 410da1efb1eec787560330ac2620892970351162. The samples build and link just fine now (both locally and in Travis).

**Comment by Armagetron** [comment]
I was linking against the updated lib and the system include files. Now I use the correct include files. Thanks!
